taskvine/src/worker/vine_sandbox.c

```diff
@@ -59,6 +59,28 @@ static int ensure_input_file( struct vine_process *p, struct vine_mount *m, stru
 	return result;
 }
 
+/*
+Ensures that each input file is present.
+*/
+
+int vine_sandbox_ensure(struct vine_process *p, struct vine_cache *cache, struct link *manager)
```

Comment:
This function currently returns a boolean, presumably true if all the files are present.
But I think there are three possibilities:
1 - All files are downloaded and present.  The task should run now.
2 - Some files have not been downloaded yet.  The task should not run yet.
3 - One or more files failed to download.  The task should be returned as a failure.

So this function needs a better return value that is handled properly where it is called.

Reply:
Or possibly case 3 should allow the task to proceed to execution, and then fail in `vine_sandbox_stagein`.